generator.py

```diff
@@ -408,66 +167,74 @@ def generate_samples(grammar_dir, outfiles):
     jsgrammar.add_import('cssgrammar', cssgrammar)
 
     for outfile in outfiles:
-        result = generate_new_sample(template, htmlgrammar, cssgrammar,
-                                     jsgrammar)
-
-        if result is not None:
-            print('Writing a sample to ' + outfile)
-            try:
-                f = open(outfile, 'w')
+        result = generate_new_sample(template, htmlgrammar, cssgrammar, jsgrammar)
+        print('Writing a sample to ' + outfile)
+        try:
+            with open(outfile, 'w') as f:
                 f.write(result)
                 f.close()
-            except IOError:
-                print('Error writing to output')
+        except IOError:
+            print('Error writing to output')
 
+def get_argument_parser():
+    
+    parser = argparse.ArgumentParser(description="DOMATO (A DOM FUZZER)")
+    
+    parser.add_argument("-f", "--file", 
+    help="File name which is to be generated in the same directory")
 
-def get_option(option_name):
-    for i in range(len(sys.argv)):
-        if (sys.argv[i] == option_name) and ((i + 1) < len(sys.argv)):
-            return sys.argv[i + 1]
-        elif sys.argv[i].startswith(option_name + '='):
-            return sys.argv[i][len(option_name) + 1:]
-    return None
+    parser.add_argument('-o', '--output_dir', type=str,
+                    help='The output directory to put the generated files in')
 
+    parser.add_argument('-n', '--no_of_files', type=int,
+                    help='number of files to be generated')
 
-def main():
-    fuzzer_dir = os.path.dirname(__file__)
-
-    multiple_samples = False
-
-    for a in sys.argv:
-        if a.startswith('--output_dir='):
-            multiple_samples = True
+    return parser
 
-    if '--output_dir' in sys.argv:
-        multiple_samples = True
+def main():
 
-    if multiple_samples:
-        print('Running on ClusterFuzz')
-        out_dir = get_option('--output_dir')
-        nsamples = int(get_option('--no_of_files'))
-        print('Output directory: ' + out_dir)
-        print('Number of samples: ' + str(nsamples))
+    fuzzer_dir = os.path.dirname(__file__)
 
-        if not os.path.exists(out_dir):
-            os.mkdir(out_dir)
+    with open(os.path.join(fuzzer_dir, "template.html"), "r") as f:
+            template = f.read()
+            f.close()
 
-        outfiles = []
-        for i in range(nsamples):
-            outfiles.append(os.path.join(out_dir, 'fuzz-' + str(i).zfill(5) + '.html'))
+    parser = get_argument_parser()
+    
+    args = parser.parse_args()
 
-        generate_samples(fuzzer_dir, outfiles)
+    if args.file:
```

Comment:
Currently this makes the behavior different than the documentation (according to the readme 'python generator.py <output file>' should work, but it doesn't anymore). Either the documentation needs to change to match the behavior or the other way around.

Reply:
mind if instead of giving the whole usage information i do something like 
```
#### Usage

To see the usage information run the following command:
`python3 generator.py --help`
```
or do i have to just add the `--file option` in the docs?

Reply:
I like to have a "quick start" command in the documentation. But feel free to include the above as well.